### PR TITLE
Delete the v1.x line from the default branch

### PR DESCRIPTION
**Pull request by steveluscher**
# Summary

In this PR we eliminate the v1.x line from the default branch, it now lives on the `maintenance/v1.x` branch.

# Test Plan

```
pnpm i
pnpm test:live-with-test-validator:setup
./scripts/start-shared-test-validator.sh
pnpm build
```

### CODE REVIEW

**Comment by steveluscher** [comment]
* **#3011** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3011?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .changeset/config.json [hunk 1]
```diff
@@ -9,12 +9,9 @@
       "repo": "solana-labs/solana-web3.js"
     }
   ],
-  "commit": false,
   "fixed": [["@solana/!({*-impl,build-scripts,test-*,tsconfig})"]],
-  "ignore": ["@solana/web3.js"],
   "linked": [],
   "snapshot": {
-    "prereleaseTemplate": "{tag}.4.{datetime}.{commit}",
```
Comment (steveluscher) [inline .changeset/config.json:17]: Changing the ‘snapshot’ version format to `2.0.0-canary.DATETIME`

File: .github/dependabot.yml [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 version: 2
 updates:
-  - package-ecosystem: 'npm'
+  - target-branch: 'maintenance/v1.x'
```
Comment (steveluscher) [inline .github/dependabot.yml:8]: Created two separate dependabot configs for the 1.x and the development branches.

File: .github/workflows/publish-canary-releases.yml [hunk 3]
```diff
@@ -32,27 +32,27 @@ jobs:
...
-      - name: Publish Snapshots
+      - name: Publish Canary Releases
         run: |
           pnpm changeset pre exit
           find packages/* -maxdepth 0 -type d -print0 | \
             xargs -t0 -n 1 -I {} \
               sh -c 'cd {} && pnpm pkg delete devDependencies'
-          pnpm changeset version --snapshot preview
-          pnpm turbo publish-packages --concurrency=95.84% --filter=\!@solana/web3.js
+          pnpm changeset version --snapshot canary
```
Comment (steveluscher) [inline .github/workflows/publish-canary-releases.yml:49]: Going forward, publishing `canary` snapshot releases on every commit.

File: .github/workflows/publish-gh-pages.yml [hunk 4]
```diff
@@ -18,7 +20,32 @@ env:
   TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
 
 jobs:
-  compile-github-pages-and-publish:
+  compile-v1-docs:
```
Comment (steveluscher) [inline .github/workflows/publish-gh-pages.yml:23]: This new GitHub pages script is supposed to:

* checkout the `maintenance/v1.x` branch, build the docs, and upload them as a built artifact
* checkout the default branch, build all the rest of the things under `compile:ghpages`, and upload them as a built artifact
* download the build artifacts and publish

File: packages/compat/package.json [hunk 5]
```diff
@@ -74,7 +74,7 @@
     },
     "devDependencies": {
         "@solana/keys": "workspace:*",
-        "@solana/web3.js": "workspace:../library-legacy"
+        "@solana/web3.js": "^1"
```
Comment (steveluscher) [inline packages/compat/package.json:77]: Locked `@solana/compat` now to the latest in the `1.x` line.

File: packages/library/README.md [hunk 6]
```diff
@@ -6,861 +6,57 @@
...
 [semantic-release-url]: https://github.com/semantic-release/semantic-release
 
-# Solana JavaScript SDK 2.0 Release Candidate
+# @solana/web3.js
 
-If you build JavaScript applications on Solana, it’s likely you’ve worked with `@solana/web3.js` or a library powered by it. With 350K+ weekly downloads on npm, it’s the most-used library in the ecosystem for building program clients, web applications, block explorers, and more.
+This is the JavaScript SDK for building Solana apps for Node, web, and React Native.
 
-Here’s an example of a common code snippet from `@solana/web3.js`:
+## Functions
```
Comment (steveluscher) [inline packages/library/README.md:19]: Wrote new README content for just the functions specific to `@solana/web3.js`.

File: README.md [hunk 7]
```diff
@@ -6,166 +6,1413 @@
...
+[npm-downloads-image]: https://img.shields.io/npm/dm/@solana/web3.js/preview.svg?style=flat
+[npm-image]: https://img.shields.io/npm/v/@solana/web3.js/preview.svg?style=flat
+[npm-url]: https://www.npmjs.com/package/@solana/web3.js/v/preview
 [semantic-release-image]: https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg
 [semantic-release-url]: https://github.com/semantic-release/semantic-release
 
 # Solana JavaScript SDK
 
-Use this to interact with accounts and programs on the Solana network through the Solana [JSON RPC API](https://solana.com/docs/rpc).
+This is the JavaScript SDK for building Solana apps for Node, web, and React Native.
```
Comment (steveluscher) [inline README.md:17]: Moved the ‘oh hi welcome here's more information than you expected’ README to the root.

File: packages/library/package.json [hunk 8]
```diff
@@ -40,8 +40,8 @@
         "compile:typedefs": "tsc -p ./tsconfig.declarations.json",
         "dev": "jest -c ../../node_modules/@solana/test-config/jest-dev.config.ts --rootDir . --watch",
         "prepublishOnly": "pnpm pkg delete devDependencies",
-        "publish-impl": "npm view $npm_package_name@$npm_package_version > /dev/null 2>&1 || pnpm publish --tag ${PUBLISH_TAG:-preview} --access public --no-git-checks",
-        "publish-packages": "sed -i 's/@solana\\/web3\\.js-experimental/@solana\\/web3\\.js/g' package.json && sed -i 's/@solana\\/web3\\.js/@solana\\/web3\\.js-bak/g' ../library-legacy/package.json && pnpm prepublishOnly && pnpm publish-impl && git reset --hard",
```
Comment (steveluscher) [inline packages/library/package.json:44]: No longer have to tweak the package name because we're on our own branch now.

File: .mergify.yml [hunk 9]
```diff
@@ -36,9 +36,6 @@ pull_request_rules:
     conditions:
       - and:
           - status-success=all-web3-checks
-          - or:
-              - -files~=^packages/library-legacy/
-              - status-success="Validate commit message"
```
Comment (steveluscher) [inline .mergify.yml:41]: This code lives on a separate branch now so we don't have to handle it.

File: packages/build-scripts/tsup.config.package.bundled_0tnraj3g9pt.mjs
Comment (lorisleiva) [inline packages/build-scripts/tsup.config.package.bundled_0tnraj3g9pt.mjs:1]: Is this normal?

File: packages/build-scripts/tsup.config.package.bundled_4jbuspluzvk.mjs
Comment (lorisleiva) [inline packages/build-scripts/tsup.config.package.bundled_4jbuspluzvk.mjs:1]: Same here.

File: packages/library/README.md (hunk 6 above)
Comment (lorisleiva) [reply packages/library/README.md:19]: Thank you! 🫶

**Review by lorisleiva — APPROVED** [review]
Tysm for doing this! I love how everything becomes so much simpler now that we don't have two libraries with the same name within the same branch.

File: .github/workflows/publish-gh-pages.yml [hunk 10]
```diff
@@ -4,9 +4,11 @@ on:
   workflow_dispatch:
     branches:
       - master
+      - maintenance/*
   push:
     branches:
       - master
+      - maintenance/*
```
[the remark this answers is not in the document]
Comment (steveluscher) [reply .github/workflows/publish-gh-pages.yml:11]: They're not separate, I actually need them to happen together. When A is updated, regenerate GH Pages completely. When B is updated, regenerate GH Pages completely.

File: .mergify.yml (hunk 9 above)
Comment (steveluscher) [reply .mergify.yml:41]: This isn't true. I actually do need this to work, but only if the branch is `maintenance/v1.x`.

File: packages/build-scripts/tsup.config.package.bundled_0tnraj3g9pt.mjs
Comment (steveluscher) [reply packages/build-scripts/tsup.config.package.bundled_0tnraj3g9pt.mjs:1]: Damnit. Deleting!

**Pull request merged**